### PR TITLE
Assign to mnemonic settings to finalConfig

### PR DESCRIPTION
**Pull request by dapplion**

### CODE REVIEW

File: src/config/mnemonic.ts [hunk 1]
```diff
@@ -49,6 +51,10 @@ export const configExtender: ConfigExtender = (finalConfig, userConfig) => {
         mnemonic: defaultMnemonic
       }
     }
+
+    // Since finalNetwork is new reference (due to { ... }) re-assign to finalConfig
+    if (!finalConfig.networks) finalConfig.networks = {}
```
Comment (0xGabi) [inline src/config/mnemonic.ts:56]: Great catch 💪

File: src/config/mnemonic.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import path from 'path'
 import { ConfigExtender, HttpNetworkConfig } from '@nomiclabs/buidler/types'
 import { aragenMnemonic } from '../params'
 import { readJsonIfExists } from '../utils/fsUtils'
+import { isHexString } from 'ethers/utils'
```
Comment (sohkai) [inline src/config/mnemonic.ts:6]: Nitpick: we usually move external imports to be ahead of local imports (easier to see which are in our control / within the same context)

File: src/config/mnemonic.ts [hunk 3]
```diff
@@ -49,6 +51,10 @@ export const configExtender: ConfigExtender = (finalConfig, userConfig) => {
         mnemonic: defaultMnemonic
       }
     }
+
+    // Since finalNetwork is new reference (due to { ... }) re-assign to finalConfig
+    if (!finalConfig.networks) finalConfig.networks = {}
+    finalConfig.networks[networkName] = finalNetwork
   }
```
Comment (sohkai) [inline src/config/mnemonic.ts:58]: Just wondering, would be it be worth returning the `finalConfig`? I do find `configExtender()`'s API a bit... unsanitary as it writes to the passed in object as opposed to just making a clone and returning it.

Comment (0xGabi) [reply src/config/mnemonic.ts:58]: We decided to use `configExtener()` with the pattern chosen by the builder team to create the `BuidlerConfig` object. They use the `ConfigExtender` type that has the following types:
```
type ConfigExtender = (config: ResolvedBuidlerConfig, userConfig: DeepReadonlyObject<BuidlerConfig>) => void
```

File: src/config/mnemonic.ts (hunk 2 above)
Comment (dapplion) [reply src/config/mnemonic.ts:6]: Fixed by https://github.com/aragon/buidler-aragon/pull/115

File: src/config/mnemonic.ts (hunk 3 above)
Comment (sohkai) [reply src/config/mnemonic.ts:58]: Ah cool, yes, makes sense to stick with their pattern!

If that type isn't coming directly from a buidler import, could we leave a note that we're mimicking their "canonical" style here?

Comment (0xGabi) [reply src/config/mnemonic.ts:58]: Everywhere we are using it we export the `ConfigExtender` type from builder. I think is quite straight forward already.

**Pull request merged**